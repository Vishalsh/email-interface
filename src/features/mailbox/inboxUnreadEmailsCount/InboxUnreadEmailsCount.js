import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Badge from 'components/badge/Badge';
import inboxUnreadEmailsCountSelector from './inboxUnreadEmailsCountSelector';
import classes from './InboxUnreadEmailsCount.module.scss';

export const InboxUnreadEmailsCount = (props) => {
  const {
    readEmailsCount,
    unreadEmailsCount,
    showReadMailsCount,
    showBadge
  } = props;

  const label = showReadMailsCount ? `${unreadEmailsCount}/${readEmailsCount}` : unreadEmailsCount;

  return (
    unreadEmailsCount !== 0 &&
    <>
    {
      showBadge ?
        <Badge className={classes.inboxUnreadEmailsCount} label={label}/>
        :
        `(${label})`
    }
    </>
  );
};

InboxUnreadEmailsCount.defaultProps = {
  showReadMailsCount: false,
  showBadge: true
};

InboxUnreadEmailsCount.propTypes = {
  readEmailsCount: PropTypes.number.isRequired,
  unreadEmailsCount: PropTypes.number.isRequired,
  showReadMailsCount: PropTypes.bool,
  showBadge: PropTypes.bool
};

const mapStateToProps = (store) => ({
  readEmailsCount: store.mailbox.inbox ? store.mailbox.inbox.emails.length : 0,
  unreadEmailsCount: inboxUnreadEmailsCountSelector(store),
});

export default connect(mapStateToProps)(InboxUnreadEmailsCount);
