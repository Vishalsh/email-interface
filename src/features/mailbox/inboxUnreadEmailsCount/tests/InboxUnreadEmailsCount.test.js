import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { InboxUnreadEmailsCount } from '../InboxUnreadEmailsCount';

describe('InboxUnreadEmailsCount', () => {
  it('should render the component when showReadMailsCount is false', () => {
    const props = {
      readEmailsCount: 10,
      unreadEmailsCount: 7,
      showReadMailsCount: false
    };
    const component = shallow(<InboxUnreadEmailsCount {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the component when showReadMailsCount is true', () => {
    const props = {
      readEmailsCount: 10,
      unreadEmailsCount: 7,
      showReadMailsCount: true
    };
    const component = shallow(<InboxUnreadEmailsCount {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the component when unreadEmailsCount is 0', () => {
    const props = {
      readEmailsCount: 10,
      unreadEmailsCount: 0,
      showReadMailsCount: false
    };
    const component = shallow(<InboxUnreadEmailsCount {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the component when showBadge is false', () => {
    const props = {
      readEmailsCount: 10,
      unreadEmailsCount: 5,
      showBadge: false
    };
    const component = shallow(<InboxUnreadEmailsCount {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});