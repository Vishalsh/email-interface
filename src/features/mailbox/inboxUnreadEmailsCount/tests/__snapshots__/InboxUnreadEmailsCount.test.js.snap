// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxUnreadEmailsCount should render the component when showBadge is false 1`] = `
<Fragment>
  (5)
</Fragment>
`;

exports[`InboxUnreadEmailsCount should render the component when showReadMailsCount is false 1`] = `
<Fragment>
  <Badge
    className="inboxUnreadEmailsCount"
    label={7}
  />
</Fragment>
`;

exports[`InboxUnreadEmailsCount should render the component when showReadMailsCount is true 1`] = `
<Fragment>
  <Badge
    className="inboxUnreadEmailsCount"
    label="7/10"
  />
</Fragment>
`;

exports[`InboxUnreadEmailsCount should render the component when unreadEmailsCount is 0 1`] = `""`;
